common/beerocks/bwl/dwpal/base_wlan_hal_dwpal.cpp

```diff
@@ -136,6 +138,21 @@ bool base_wlan_hal_dwpal::fsm_setup()
                     attached = true;
                 }
 
+                //Attach NL interface
+                if (m_dwpal_nl_ctx == nullptr) {
+                    //Attach nl interface
+                    if (dwpal_driver_nl_attach(&m_dwpal_nl_ctx) == DWPAL_FAILURE) {
+                        LOG(ERROR) << "dwpal_driver_nl_attach returned ERROR for interface ="
+                                   << m_radio_info.iface_name;
+                        return (transition.change_destination(dwpal_fsm_state::Detach));
```

Comment:
The reason that i initially thought of was just not to crash in case platform dosnt support NL , but then I see now that its not possible . 
On the other hand i dont see a point running with ctrl interfaces but without NL , so i think we should set attach = false in case of nl attach failure as well .
@arnout what do you think?

Reply:
That's exactly what I meant in https://github.com/prplfoundation/prplMesh/pull/619#discussion_r362031408

Reply:
so, to conclude the changes we agree on:
1. remove the "attached = true;" from the else case.
2. add "attach = false" in case dwpal_driver_nl_attach() fails

Reply:
added the attach = false to the attach() fail scenario